apps/web/src/stores/documentVersions.ts

```diff
@@ -0,0 +1,118 @@
+'use client'
+
+import { useCallback } from 'react'
+
+import { DocumentVersion } from '@latitude-data/core'
+import {
+  SidebarDocument,
+  useCurrentCommit,
+  useCurrentProject,
+  useToast,
+} from '@latitude-data/web-ui'
+import { createDocumentVersionAction } from '$/actions/documents/create'
+import { destroyDocumentAction } from '$/actions/documents/destroyDocumentAction'
+import { getDocumentsAtCommitAction } from '$/actions/documents/getDocumentsAtCommitAction'
+import { ROUTES } from '$/services/routes'
+import { useRouter } from 'next/navigation'
+import useSWR, { SWRConfiguration } from 'swr'
+import { useServerAction } from 'zsa-react'
+
+export default function useDocumentVersions(
+  { currentDocument }: { currentDocument: SidebarDocument | undefined },
+  opts?: SWRConfiguration,
+) {
+  const { toast } = useToast()
+  const router = useRouter()
+  const { project } = useCurrentProject()
+  const { commit } = useCurrentCommit()
+  const { mutate, data, ...rest } = useSWR<DocumentVersion[]>(
+    ['documentVersions', project.id, commit.id],
```

Comment:
Let's agree on using Arrays for keys in the stores. URL like strings are harder to manipulate